backend/inspirehep/disambiguation/tasks.py

```diff
@@ -206,32 +239,21 @@ def disambiguate_authors(self, record_uuid):
     for author in authors:
         if author.get("curated_relation"):
             continue
-        matched_authors = list(
-            match(author, current_app.config["AUTHOR_MATCHER_EXACT_CONFIG"])
-        )
-        matched_authors_refs = {
-            matched_author["_source"]["self"]["$ref"]
-            for matched_author in matched_authors
-        }
-        if len(matched_authors_refs) == 1:
-            updated_author_recid = assign_reference_to_author_if_unambiguous_match(
-                matched_authors_refs, author
+        author_bai = remove_author_bai_from_id_list(author)
+        updated_author_recid = match_author(author, updated_authors)
+        if not updated_author_recid:
+            updated_author_recid = match_literature_author(
+                author, updated_authors, record
             )
```

Comment:
are you populating the BAI in this case? doesn't look like it. 

Additionally, I think this function is slowly becoming a big mess. From a high-level perspective, what you need to do here is:
1. match the current record against author records, ignoring the BAI among IDs,
2.  if there is no result, match against literature records,
3. and if still no results and there was no assigned record yet, create a new record
4. in all cases, you need to populate the signature with the record ref and BAI.

Does the code match this high-level description? not really, as e.g. 1. would be `match_author`, but that function does both more and less: it doesn't take care of the BAI (you need to do that separately before in an awkward way that requires you to restore it back at the end, which is error-prone and mixing different levels of concern), but OTOH it returns results through mutation of the underlying record (for the BAI) and through a return value (for the recid). That's nasty! Instead, it should 
* internally deal with removing the BAI, without that concern being visible to the outside world  (this function) as it's a detail of how we match authors.
* return both BAI and recid (through a `tuple` or a `dict`, up to you).

Same pattern for `match_literature_author` which should only match and return both pieces of info, then `create_new_author`. Then once at the end, you can check if you have a new BAI and record ID, and put them into the signature if needed.